chain/beacon/store_test.go

```diff
@@ -42,9 +42,9 @@ func TestSchemeStore(t *testing.T) {
 
 	// test if store sets to nil prev signature depending on scheme
 	// with chained scheme, it should keep the consistency between prev signature and signature
-	if sch.DecouplePrevSig && beaconSaved.PreviousSig != nil {
+	if !sch.IsPreviousSigSignificant && beaconSaved.PreviousSig != nil {
```

Comment:
a more extensible way to achieve the same could be to provide a `Verify: func(Beacon) bool` to the Scheme config

Reply:
rather than `IsPreviousSigSignificant` that is

Reply:
I'll probably do that, yeah